securedrop/tests/functional/functional_test.py

```diff
@@ -33,20 +34,69 @@ class FunctionalTest():
 
     def _unused_port(self):
         s = socket.socket()
-        s.bind(("localhost", 0))
+        s.bind(('127.0.0.1', 0))
         port = s.getsockname()[1]
         s.close()
         return port
 
-    def _create_webdriver(self):
-        log_file = open(join(LOG_DIR, 'firefox.log'), 'a')
-        log_file.write(
-            '\n\n[%s] Running Functional Tests\n' % str(
-                datetime.now()))
-        log_file.flush()
+
+    def _create_tor_browser_webdriver(self, abs_log_file_path):
+        # Creating the TorBrowser driver changes the working directory,
+        # so we need to save and then restore the current one.
+        old_dir = os.getcwd()
+
+        # Read the TorBrowser location from Ansible
+        with open('/opt/.tbb_path_file') as f:
+           path_to_tbb = f.readline().strip()
+        path_to_tbb = path_to_tbb + os.path.sep + "tor-browser_en-US"
+        path_to_tbb = os.path.abspath(path_to_tbb)
+
+        # Due to Travis having opinions, we need to chown the tor dir
+        # to be owned by root when we're running CI.
```

Comment:
This is much better done by Ansible. First, make the change I recommend regarding removing the system account use above, then see if this is still needed. If it is, then `become: True` in Ansible when running under Travis _and_ in the relevant Ansible explain exactly why it's necessary (more info than "Travis has opinions" would be helpful :smile:).